pkg/codegen/options.go

```diff
@@ -0,0 +1,122 @@
+package codegen
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
+	"text/template"
+
+	"gopkg.in/yaml.v3"
+)
+
+type (
+	optionsDef struct {
+		Source    string
+		outputDir string
+
+		Name string
+
+		// List of imports
+		// Used only by generated file and not pre-generated-user-file
+		Imports []string `yaml:"imports"`
+
+		Properties optionsPropSet `yaml:"props"`
+	}
+
+	optionsPropSet []*optionsProp
+
+	optionsProp struct {
+		Name    string
+		Type    string
+		Env     string
+		Default *optionsPropDefault
+	}
+
+	optionsPropDefault string
+)
+
+// Processes multiple options defenitions
+func procOptions(mm ...string) (dd []*optionsDef, err error) {
+	var (
+		f io.ReadCloser
+		d *optionsDef
+	)
+
+	dd = make([]*optionsDef, 0)
+	for _, m := range mm {
+		err = func() error {
+			if f, err = os.Open(m); err != nil {
+				return err
+			}
+
+			defer f.Close()
+
+			d = &optionsDef{}
+
+			if err := yaml.NewDecoder(f).Decode(d); err != nil {
+				return err
+			}
+
+			for _, j := range d.Properties {
+
+				if j.Type == "" {
+					j.Type = "string"
+				}
+
+				if j.Env == "" {
+					j.Env = strings.ToUpper(d.Name + "_" + cc2underscore(j.Name))
+				}
+
+			}
+
+			d.Source = m
+			d.outputDir = path.Dir(m)
+
+			dd = append(dd, d)
+
+			return nil
+		}()
+		if err != nil {
+			return nil, fmt.Errorf("could not process %s: %w", m, err)
+		}
+	}
+	return dd, nil
+}
+
+// Custom UnmarshelYAML function for
```

Comment:
Typo :D 
I see no other problems, that's why I'm pointing this trivial one out.
Great job 👍